src/test/java/org/mockito/internal/configuration/plugins/PluginLoaderTest.java

```diff
@@ -0,0 +1,88 @@
+package org.mockito.internal.configuration.plugins;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.internal.util.io.IOUtil;
+import org.mockito.plugins.PluginSwitch;
+import org.mockitoutil.ClassLoaders;
+import org.mockitoutil.TestBase;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.anyString;
+
+/**
+ *  Test PluginLoader for case when Plugin class is loaded via class loader which is different from current context
+ *  class loader.
+ */
+public class PluginLoaderTest extends TestBase {
+
+    @InjectMocks
+    private PluginLoader pluginLoader;
+
+    @Mock
+    private PluginSwitch pluginSwitch;
+
+    @Before
+    public void setUp() {
+        given(pluginSwitch.isEnabled(anyString())).willReturn(true);
+    }
+
+    @Test
+    public void without_custom_context_class_loader_plugin_should_be_load() {
+
+
+        Object pluginInterface = pluginLoader.loadPlugin(PluginInterface.class, "org.mockito.internal" +
+                                                                                                  ".configuration.plugins.DefaultPlugin");
+
+        assertEquals(PluginImpl.class, pluginInterface.getClass());
+    }
+
+    @Test
+    public void if_custom_context_class_loader_is_used_then_plugin_should_be_load() throws Exception {
+
+        ClassLoader classLoader = ClassLoaders.inMemoryClassLoader().withParent(getParentClassLoader()).withClassDefinition("", readResource()).build();
+
+        Thread.currentThread().setContextClassLoader(classLoader);
+
+        Object pluginInterface = pluginLoader.loadPlugin(PluginInterface.class, "org.mockito.internal" +
+                                                                                                  ".configuration.plugins.DefaultPlugin");
+
+        assertEquals(PluginImpl.class, pluginInterface.getClass());
```

Comment:
Suggested assert will be incorrect.  The `PluginImpl`should be loaded via `classLoader` (who plays role context class loader). It should be loaded with `AppClassLoader` to be assignable to fields of `PluginRegistry` even if context class loader != `AppClassLoader`. I change assert to more readable.